sway-core/src/semantic_analysis/ast_node/declaration/monomorphize.rs

```diff
@@ -0,0 +1,12 @@
+use sway_types::{Ident, Span};
+
+use crate::{namespace::Items, semantic_analysis::TypeMapping, TypeParameter};
+
+pub(crate) trait MonomorphizeHelper {
+    type Output;
+
+    fn type_parameters(&self) -> &[TypeParameter];
+    fn name(&self) -> &Ident;
+    fn span(&self) -> &Span;
```

Comment:
Are the `type_parameters`, `name` or `span` methods used anywhere?

Reply:
They are used in the PR following this one. I added them here to make that change more clear.